micrometer-test/src/main/java/io/micrometer/core/instrument/InstrumentationVerificationTests.java

```diff
@@ -29,4 +49,142 @@ protected MeterRegistry getRegistry() {
         return registry;
     }
 
+    /**
+     * Helper method for creating a {@link TestObservationRegistry} to run the tests also
+     * using the Observation API.
+     * @return a {@link TestObservationRegistry} with a
+     * {@link DefaultMeterObservationHandler} registered
+     */
+    protected TestObservationRegistry createObservationRegistryWithMetrics() {
+        TestObservationRegistry observationRegistry = TestObservationRegistry.create();
+        observationRegistry.observationConfig().observationHandler(new DefaultMeterObservationHandler(getRegistry()));
+        return observationRegistry;
+    }
+
+    protected ObservationRegistry getObservationRegistry() {
+        return this.testObservationRegistry;
+    }
+
+    /**
+     * Enum that represents types of tests we can run.
+     */
+    enum TestType {
+
+        /**
+         * Runs the tests by just using the {@link MeterRegistry}.
+         */
+        METRICS_VIA_METER_REGISTRY,
+
+        /**
+         * Runs the tests by using the Observation API.
+         */
+        METRICS_VIA_OBSERVATIONS_WITH_METRICS_HANDLER
+
+    }
+
+    // Allows parameter resolution in BeforeEach and AfterEach
+    // Based on
+    // https://code-case.hashnode.dev/how-to-pass-parameterized-test-parameters-to-beforeeachaftereach-method-in-junit5
+    static class AfterBeforeParameterResolver implements BeforeEachMethodAdapter, ParameterResolver {
```

Comment:
It's a shame we need to maintain this code here, but unless https://github.com/junit-team/junit5/issues/944 is implemented, I guess we don't have much of a choice.